### ISSUE TITLE
onItemPress: put navigation code in setTimeout

### ISSUE THREAD

**Issue by Akshg** [react-navigation/react-navigation#2608]
My drawer was lagging. So I wrote the following code in onItemPress:

```
<DrawerItems 
          {...props}
          onItemPress={({ route, focused }) => {
            if(!focused) {
              setTimeout(() => {              
                props.navigation.navigate(route.routeName)
              }, 0)
            }
            props.navigation.navigate('DrawerClose');
          }}
/>
```

I suggest that if we put the navigation part in setTimeout then performance will increase.

 ```
 _onItemPress = ({ route, focused }: DrawerItem) => {
    this.props.navigation.navigate('DrawerClose');
    if (!focused) {
      setTimeout(() => {              
          let subAction;
          // if the child screen is a StackRouter then always navigate to its first screen (see #1914)
          if (route.index !== undefined && route.index !== 0) {
            route = ((route: any): NavigationStateRoute);
            subAction = NavigationActions.navigate({
              routeName: route.routes[0].routeName,
            });
         }
         this.props.navigation.navigate(route.routeName, undefined, subAction);
      }, 0)
    }
  };
```

**Comment by spencercarli** [comment]
This wouldn't be the right solution to this problem. It would be better to determine the root cause of the slow drawer. Can you put together a sample app that demonstrates the issue?

**Comment by akash-cp** [comment]
My package.json:
```
{
  "name": "ClassproStudentApp",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "eject-native-base": "node node_modules/native-base/ejectTheme.js"
  },
  "dependencies": {
    "filesize": "3.3.0",
    "lodash": "4.15.0",
    "moment": "2.14.1",
    "native-base": "^2.1.1",
    "react": "16.0.0-alpha.6",
    "react-native": "0.43.3",
    "react-native-fabric": "^0.4.1",
    "react-native-fbsdk": "^0.5.0",
    "react-native-fcm": "^6.2.2",
    "react-native-modal-datetime-picker": "^4.2.1",
    "react-redux": "^5.0.4",
    "redux": "^3.6.0",
    "redux-logger": "^3.0.1",
    "redux-persist": "^4.8.0",
    "redux-promise-middleware": "^4.2.0",
    "redux-thunk": "^2.2.0"
  }
}

```
Handset: Xiaomi redmi 2

You can see the difference here.
Before: https://drive.google.com/file/d/0B0qI7gXjfrKkNE15N2psWUJZcVU
After: https://drive.google.com/open?id=0B0qI7gXjfrKkUEU5WlE1V1MtOVk

I'll post sample app shortly.

**Comment by geekvijay** [comment]
@spencercarli, One reason could be that navigation to other route is not waiting for drawer closing animation to get completed and that's why it is lagging.

**Comment by kelset** [comment]
Pinging OP @Akshg  since this issue has not been active for a while, and it's related to an old version of the lib.

Please let me know if you want this to remain open (does it still happen in the latest version?); if I get no answer in the next 7 days I will close it.

**Comment by kelset** [comment]
Hi there @Akshg,

In an effort to cleanup this project and prioritize a bit, since this issue had no follow up since my last comment I'm going to close it.

If you are still having the issue with the latest version (especially if it's a bug report) please check the current open issues, and if you can't find a duplicate, open a new one that uses the new Issue Template to provide some more details to help us solve it.
